### PR TITLE
Ts types improvements

### PR DESCRIPTION
**Pull request by hiredgun**
This PR improve types and adds misc minor improvements

### CODE REVIEW

File: src/number-formatting/index.ts [hunk 1]
```diff
@@ -111,28 +122,30 @@ class NumberFormatting implements NumberFormattingOptions {
...
 
     /**
      * Returns the actual number of decimals that a number has.
-     * @param number - number
-     *
+     * @param number - number or numeric string
      */
     getActualDecimals(number: number) {
-        number = Math.abs(number);
+        number = Math.abs(Number(number));
```
Comment (lukeapage) [inline src/number-formatting/index.ts:148]: Do you need to do it here? I see it being done on the callsites too..

File: src/utils/fetch.ts [hunk 2]
```diff
@@ -1,36 +1,11 @@
...
+import type { HTTPMethodType, NetworkError, OAPIRequestResult } from '../types';
 
 interface Options {
     /**
-     *  The body of the request. If this is an object, that is not already handled by the body mixin
-     *  it is converted to JSON and the appropriate content-type header added.
+     *  The body of the request.
      */
-    body?: BodyInit | Record<string, unknown>;
+    body?: any;
```
Comment (lukeapage) [inline src/utils/fetch.ts:8]: do we really allow anything?

**Comment by lukeapage** [comment]
Added 2 comments

File: src/number-formatting/index.ts (hunk 1 above)
Comment (hiredgun) [reply src/number-formatting/index.ts:148]: Ahh, yeah, thanks, I allowed strings before so it's not needed now.

File: src/utils/fetch.ts (hunk 2 above)
Comment (hiredgun) [reply src/utils/fetch.ts:8]: Theoretically yes, our fetch wrapper passes everything not compatible with BodyInit (type expected by original fetch) through `JSON.stringify` which accepts anything and returns `string|undefined`.

**Pull request merged**